libraries/chain/include/eosio/chain/abi_serializer.hpp

```diff
@@ -577,12 +583,74 @@ namespace impl {
             }
          } catch(...) {}
 
-         mvo("call_traces", act_trace.call_traces);
+         add(mvo, "call_traces", act_trace.call_traces, resolver, ctx);
          mvo("console_markers", act_trace.console_markers);
 
          out(name, std::move(mvo));
       }
 
+      /**
+       * overload of to_variant_object for call_trace
+       *
+       * This matches the FC_REFLECT for this type, but this is provided to extract the contents of call_trace.data and call_trace.return_value
+       * @tparam Resolver
+       * @param call_trace
+       * @param resolver
+       * @return
+       */
+      template<typename Resolver>
+      static void add( mutable_variant_object& out, const char* name, const call_trace& cal_trace, const Resolver& resolver, abi_traverse_context& ctx )
+      {
+         static_assert(fc::reflector<call_trace>::total_member_count == 12);
+         auto h = ctx.enter_scope();
+         mutable_variant_object mvo;
+
+         mvo("call_ordinal", cal_trace.call_ordinal);
+         mvo("sender_ordinal", cal_trace.sender_ordinal);
+         mvo("receiver", cal_trace.receiver);
+         mvo("read_only", cal_trace.read_only);
+
+         call_data_header data_header;
+         fc::datastream<const char*> ds( cal_trace.data.data(), cal_trace.data.size() );
+         fc::raw::unpack( ds, data_header );
+         std::string fname = eosio::chain::name(data_header.func_name).to_string();
+
+         mvo("raw_data", cal_trace.data);
+         try {
+            auto abi_optional = resolver(cal_trace.receiver);
+            if (abi_optional) {
+               const abi_serializer& abi = *abi_optional;
+               auto type = abi.get_call_type(fname);
+               if (!type.empty()) {
+                  call_data_to_variant_context _ctx(abi, ctx, type);
+                  mvo( "unpacked_data", abi._binary_to_variant( type, cal_trace.data, _ctx ));
+               }
+            }
+         } catch(...) {}
+
+         mvo("elapsed", cal_trace.elapsed);
+         mvo("console", cal_trace.console);
+         mvo("console_markers", cal_trace.console_markers);
```

Comment:
I don't remember. Were we going to expand these out in anyway?

Reply:
Thanks. I was planning to do it while doing pretty printing. But will do it now.